src/main/java/com/lunark/lunark/reservations/service/ReservationService.java

```diff
@@ -89,6 +93,72 @@ public List<Reservation> getAllReservationsForPropertiesList(List<Property> prop
         return reservationsForProperties;
     }
 
+    @Override
+    public List<Reservation> getIncomingReservationsForHostId(Long hostId) {
+        List<Property> properties = propertyRepository.findAll().stream().filter(property -> Objects.equals(property.getHost().getId(), hostId)).toList();
+        List<Reservation> reservationsForProperties = new ArrayList<>();
+        for(Property property: properties) {
+            List<Reservation> reservationForProperty = reservationRepository.findByPropertyId(property.getId());
+            reservationsForProperties.addAll(reservationForProperty);
+        }
+        return reservationsForProperties;
+    }
+
+    @Override
+    public List<Reservation> getAllAcceptedReservations(Long guestId) {
+        return reservationRepository.findAll().stream().filter(reservation -> Objects.equals(reservation.getGuest().getId(), guestId) && reservation.getStatus() == ReservationStatus.ACCEPTED).toList();
+    }
+
+    @Override
+    public void save(Reservation reservation) {
+        Optional<Reservation> reservationUpdate = findById(reservation.getId());
+        if (reservationUpdate.isPresent()) {
+            Reservation existingReservation = reservationUpdate.get();
+            existingReservation.copyFields(reservation);
+            reservationRepository.saveAndFlush(existingReservation);
+        } else {
+            throw new RuntimeException("Reservation not found with id: " + reservation.getId());
+        }
+    }
+
+    @Override
+    public void updateReservations(Reservation reservation) {
+        Long propertyId = reservation.getProperty().getId();
+        List<Reservation> allPropertyReservations = reservationRepository.findByPropertyId(propertyId);
+
+        for (Reservation existingReservation : allPropertyReservations) {
+            if (doDatesOverlap(reservation, existingReservation)) {
```

Comment:
The reservation that was just accepted will be in `allPropertyReservations`. As the reservation overlaps with itself, it will end up being cancelled here.